management/networksecuritygroup/client.go

```diff
@@ -0,0 +1,254 @@
+	deleteSecurityGroupURL           = "services/networking/networksecuritygroups/%s"
+	getSecurityGroupURL              = "services/networking/networksecuritygroups/%s?detaillevel=full"
+	listSecurityGroupsURL            = "services/networking/networksecuritygroups"
+	addSecurityGroupToSubnetURL      = "services/virtualnetwork/%s/subnets/%s/networksecuritygroups"
+	getSecurityGroupForSubnetURL     = "services/virtualnetwork/%s/subnets/%s/networksecuritygroups"
+	removeSecurityGroupFromSubnetURL = "services/virtualnetwork/%s/subnets/%s/networksecuritygroups/%s"
+	setSecurityGroupRuleURL          = "services/networking/networksecuritygroups/%s/rules/%s"
+	deleteSecurityGroupRuleURL       = "services/networking/networksecuritygroups/%s/rules/%s"
+
+	errParamNotSpecified = "Parameter %s is not specified."
+)
+
+// NewClient is used to instantiate a new SecurityGroupClient from an Azure client
+func NewClient(client management.Client) SecurityGroupClient {
+	return SecurityGroupClient{client: client}
+}
+
+// CreateNetworkSecurityGroup creates a new network security group within
+// the context of the specified subscription
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn913818.aspx
+func (sg SecurityGroupClient) CreateNetworkSecurityGroup(
+	name string,
+	label string,
+	location string) (management.OperationId, error) {
+	if name == "" {
+		return "", fmt.Errorf(errParamNotSpecified, "name")
+	}
+	if location == "" {
+		return "", fmt.Errorf(errParamNotSpecified, "location")
+	}
+
+	data, err := xml.Marshal(SecurityGroupRequest{
+		Name:     name,
+		Label:    label,
+		Location: location,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	requestURL := fmt.Sprintf(createSecurityGroupURL)
+	return sg.client.SendAzurePostRequest(requestURL, data)
+}
+
+// DeleteNetworkSecurityGroup deletes the specified network security group from the subscription
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn913825.aspx
+func (sg SecurityGroupClient) DeleteNetworkSecurityGroup(
+	name string) (management.OperationId, error) {
+	if name == "" {
+		return "", fmt.Errorf(errParamNotSpecified, "name")
+	}
+
+	requestURL := fmt.Sprintf(deleteSecurityGroupURL, name)
+	return sg.client.SendAzureDeleteRequest(requestURL)
+}
+
+// GetNetworkSecurityGroup returns information about the specified network security group
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn913821.aspx
+func (sg SecurityGroupClient) GetNetworkSecurityGroup(name string) (SecurityGroupResponse, error) {
+	if name == "" {
+		return SecurityGroupResponse{}, fmt.Errorf(errParamNotSpecified, "name")
+	}
+
+	requestURL := fmt.Sprintf(getSecurityGroupURL, name)
+	response, err := sg.client.SendAzureGetRequest(requestURL)
+	if err != nil {
+		return SecurityGroupResponse{}, err
+	}
+
+	var securityGroup SecurityGroupResponse
+	if err := xml.Unmarshal(response, &securityGroup); err != nil {
+		return SecurityGroupResponse{}, err
```

Comment:
you can return `securityGroup` here as well. (that brings me to my earlier comment, you can just do the following):

    err := xml.Unmarshal
    return obj, err

and it'll be just fine. :) I remember you had a readability argument but it seems clearer to me. Up to you if you like to change, not a big deal.